packages/common/src/cache/cache-record-schema-maker.ts

```diff
@@ -0,0 +1,27 @@
+import {App} from '@yoroi/types'
+import z, {ZodType} from 'zod'
+
+export const cacheRecordSchemaMaker = <T>(
+  recordSchema: ZodType<T>,
+): ZodType<App.CacheRecord<T>> => {
+  return z.object({
+    record: recordSchema,
+    expires: z.number().nonnegative(),
+    hash: z.string(),
+    // Zod is not infering the type correctly when using interface
+  }) as unknown as ZodType<App.CacheRecord<T>>
```

Comment:
zod loses reference when inferring with another schema as part of the current schema

Reply:
checking

Reply:
I remember I tested a lot of stuff couldn't get it working ```
Type 'ZodType<CacheRecord<T>, ZodTypeDef, CacheRecord<T>>' does not satisfy the constraint 'ZodRawShape'.
  Index signature for type 'string' is missing in type 'ZodType<CacheRecord<T>, ZodTypeDef, CacheRecord<T>>'
```